internal/pkg/remote/credential/login_handler.go

```diff
@@ -87,6 +90,14 @@ func (h *ociHandler) login(u *url.URL, username, password string, insecure bool)
 		return nil, err
 	}
 
+	if ociAuthFile != "" {
```

Comment:
There shouldn't be any need for this additional block that's bringing in `containers/image`. A custom auth file should be handled in the same flow we are using for `~/.singularity/docker-config.json`

You should be able to handle a different file through changing `ociConfig` and `ociConfigNew` below.

Reply:
Underlying cause is #2226; marking as resolved.